api/class-wc-rest-dev-orders-controller.php

```diff
@@ -28,4 +28,51 @@ class WC_REST_Dev_Orders_Controller extends WC_REST_Orders_Controller {
 	 */
 	protected $namespace = 'wc/v3';
 
+	/**
+	 * Prepare objects query.
+	 *
+	 * @since  3.0.0
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return array
+	 */
+	protected function prepare_objects_query( $request ) {
+		global $wpdb;
+
+		// This is a hack to get around WC_REST_Orders_Controller::prepare_objects_query
+		$status = $request['status'];
+		unset( $request['status'] );
+
+		$args = parent::prepare_objects_query( $request );
+
+		// Set post_status.
+		if ( isset( $status ) ) {
```

Comment:
This will always be `isset` because you set it above. You need to move the isset to check `$request['status']`